c2corg_api/tests/views/test_document_delete.py

```diff
@@ -0,0 +1,562 @@
+
+        # Check that entries have been removed in the main tables
+        count = self.session.query(clazz). \
+            filter(getattr(clazz, 'document_id') == document_id).count()
+        self.assertEqual(0, count)
+        count = self.session.query(Document). \
+            filter(Document.document_id == document_id).count()
+        self.assertEqual(0, count)
+        if clazz_locale:
+            count = self.session.query(clazz_locale). \
+                filter(getattr(clazz_locale, 'document_id') == document_id). \
+                count()
+            self.assertEqual(0, count)
+        count = self.session.query(DocumentLocale). \
+            filter(DocumentLocale.document_id == document_id).count()
+        self.assertEqual(0, count)
+        count = self.session.query(DocumentGeometry). \
+            filter(DocumentGeometry.document_id == document_id).count()
+        self.assertEqual(0, count)
+
+        # Check the archives have been cleared too
+        count = self.session.query(archive_clazz). \
+            filter(getattr(archive_clazz, 'document_id') == document_id). \
+            count()
+        self.assertEqual(0, count)
+        count = self.session.query(ArchiveDocument). \
+            filter(ArchiveDocument.document_id == document_id).count()
+        self.assertEqual(0, count)
+        if archive_clazz_locale:
+            count = self.session.query(archive_clazz_locale). \
+                filter(getattr(
+                    archive_clazz_locale, 'document_id') == document_id). \
+                count()
+            self.assertEqual(0, count)
+        count = self.session.query(ArchiveDocumentLocale). \
+            filter(ArchiveDocumentLocale.document_id == document_id).count()
+        self.assertEqual(0, count)
+        count = self.session.query(ArchiveDocumentGeometry). \
+            filter(ArchiveDocumentGeometry.document_id == document_id).count()
+        self.assertEqual(0, count)
+
+        # check associations have been cleared
+        association_count = self.session.query(Association).filter(or_(
+            Association.parent_document_id == document_id,
+            Association.child_document_id == document_id
+        )).count()
+        self.assertEqual(0, association_count)
+        association_log_count = self.session.query(AssociationLog).filter(or_(
+            AssociationLog.parent_document_id == document_id,
+            AssociationLog.child_document_id == document_id
+        )).count()
+        self.assertEqual(0, association_log_count)
+        association_count = self.session.query(TopoMapAssociation).filter(
+            TopoMapAssociation.document_id == document_id).count()
+        self.assertEqual(0, association_count)
+        association_count = self.session.query(AreaAssociation).filter(
+            AreaAssociation.document_id == document_id).count()
+        self.assertEqual(0, association_count)
+
+        # TODO check cache_versions have been updated
```

Comment:
For example when deleting `self.waypoint3` you could test that the cache version of the associated route `self.route3` has changed:

```python
        self.check_cache_version(self.route3.document_id, 2)

```